generate-test-vector.py

```diff
@@ -1241,6 +1241,51 @@ def generate_taproot_with_nums_point_test():
 
     return test_cases
 
+def generate_pay_to_multisig_test():
+    sender, recipient, test_case = new_test_case()
+    outpoint = ("f4184fc596403b9d638783cf57adfe4c75c605f6356fbc91338530e9831e9e16", 0)
+    sender_bip32_seed = 'deadbeef'
+    recipient_bip32_seed = 'f00dbabe'
+    scan, spend, Scan, Spend = reference.derive_silent_payment_key_pair(bytes.fromhex(recipient_bip32_seed))
+    address = reference.encode_silent_payment_address(Scan, Spend, hrp=HRP)
+    addresses = [(address, 1.0)]
+
+    sender['given']['recipients'] = addresses
+    recipient['given']['key_material']['scan_priv_key'] = scan.get_bytes().hex()
+    recipient['given']['key_material']['spend_priv_key'] = spend.get_bytes().hex()
+    recipient['expected']['addresses'] = [address]
+
+    keys = [
+        get_key_pair(0, seed=bytes.fromhex(sender_bip32_seed)),
+        get_key_pair(1, seed=bytes.fromhex(sender_bip32_seed))
+    ]
+    # OP_2 pub1 pub2 OP_3 OP_EQUAL
+    multisig_script = "52" + ''.join(["21"+key[1].get_bytes(False).hex() for key in keys]) + "53ae"
```

Comment:
For P2MS, you need all of the public keys in the `scriptPubKey`, so this should be `OP_2 pub1 pub2 pub2 OP_3 OP_CHECKMULTISIG`

Reply:
> For P2MS, you need all of the public keys in the `scriptPubKey`, so this should be `OP_2 pub1 pub2 pub2 OP_3 OP_CHECKMULTISIG`

I'm doing a join here over an array of pub keys so I still achieve the same result

Reply:
I think you misunderstood my point. Your array of pubkeys only has two keys in it and you are trying to create a 2-of-3 multisig. For a M-of-N multisig, you need N pubkeys in the script (3 in this case), and M signatures in the `scriptSig`. The algorithm for `OP_CHECKMULTISIG` is to put everything on the stack and then check every signature against every pubkey. The order of the signatures also matters, in that once a pubkey fails a signature check, it is not checked again for the next signature. Its a pretty janky OP_CODE, which is why OP_CHECKMULTISIG was deprecated in tapscript in favor of the new OP_CHECKSIGADD.

More details on P2MS: https://learnmeabitcoin.com/technical/p2ms

Reply:
@josibake

> For P2MS, you need all of the public keys in the `scriptPubKey`, so this should be `OP_2 pub1 pub2 pub2 OP_3 OP_CHECKMULTISIG`

I've fixed this. I forgot about it initially. It is now `OP_2 pub1 pub2 pub3 OP_3 OP_CHECKMULTISIG`. Then I use 2 signatures from first two keys in my `scriptSig`